### ISSUE TITLE
[bug] Regression in >=1.51.0: `-Werror=dev --warn-uninitialized` triggers errors

### ISSUE THREAD

**Issue by friendlyanon** [conan-io/conan#11904]
This was previously reported and fixed in https://github.com/conan-io/conan/issues/10288

https://github.com/friendlyanon/cmake-init/issues/67 reported to me that this regression happened and indeed it only happens in the versions mentioned in the title.

### Steps to reproduce (Include if Applicable)

CMakeLists.txt:

```cmake
cmake_minimum_required(VERSION 3.15)

project(proj CXX)

find_package(Catch2 REQUIRED)
```

conanfile.py:

```py
from conan import ConanFile


class Recipe(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeToolchain", "CMakeDeps", "VirtualRunEnv"

    def layout(self):
        self.folders.generators = "conan"

    def requirements(self):
        self.requires("catch2/3.0.1")
```

Run the commands:

- `pip install -U conan==1.51.0`
- `conan install . -s build_type=Debug`
- `cmake -B build --toolchain conan/conan_toolchain.cmake -Werror=dev --warn-uninitialized`

**Comment by franramirez688** [comment]
Hi @friendlyanon 

Thanks for reporting it!

This issue seems to be solved by the latest Conan 1.51.3 (https://github.com/conan-io/conan/releases/tag/1.51.3).
Let us know if it's working now to close the issue.

**Comment by friendlyanon** [comment]
Indeed, the issue is not present in 1.51.3 with the reproducer.

Thanks! That's what I call a quick fix :)
